miio/gateway.py

```
@@ -536,38 +513,216 @@ class SubDevice:
+        except Exception as ex:
+            _LOGGER.debug(
+                "Got an exception while setting propertie %s to value %s: %s", property, str(value), ex, exc_info=True
+            )
+            return None
+
+    @command()
+    def unpair(self):
+        return self.subdevice_send("remove_device")
+
+    @command()
+    def get_battery(self):
+        self._battery = self.subdevice_send("get_battery").pop()
+        return self._battery
+
+    @command()
+    def get_firmware_version(self) -> Optional[int]:
+        """Returns firmware version"""
+        return self.get_subdevice_prop("fw_ver").pop()
+
+
+class AqaraHT(SubDevice):
+    _temperature = None
+    _humidity = None
+    _pressure = None
+
+    @property
+    def temperature(self):
+        """Return the temperature in degrees celsius"""
+        return self._temperature
+
+    @property
+    def humidity(self):
+        """Return the humidity in %"""
+        return self._humidity
+
+    @property
+    def pressure(self):
+        """Return the pressure in hPa"""
+        return self._pressure
+
+    @command()
+    def update(self):
+        """Update all device properties"""
+        values = self.get_subdevice_prop_exp(["temperature", "humidity", "pressure"])
+        if len(values) == 3:
+            self._temperature = values[0]/100
+            self._humidity = values[1]/100
+            self._pressure = values[2]/100
+        return
 
 
 class SensorHT(SubDevice):
     accessor = "get_prop_sensor_ht"
     properties = ["temperature", "humidity"]
+    _temperature = None
+    _humidity = None
 
+    @property
+    def temperature(self):
+        return self._temperature
+
+    @property
+    def humidity(self):
+        return self._humidity
+    
+    @command()
+    def update(self):
+        values = self.get_subdevice_prop_exp(properties)
+        if len(values) == 2:
+            self._temperature = values[0]
+            self._humidity = values[1]
+        return
```

I'd say this should raise an exception if the response is unexpected, to avoid failing quietly.

---

done